src/sif/autocost.cc

```diff
@@ -470,7 +497,7 @@ Cost AutoCost::TransitionCost(const baldr::DirectedEdge* edge,
     // Still want to add a penalty so routes avoid high cost intersections.
     float seconds = turn_cost * edge->stopimpact(idx);
     // Apply density factor penality if there isnt traffic on this edge or youre not using traffic
-    if (!edge->has_flow_speed() || flow_mask_ == 0)
+    if (!edge->has_flow_speed() && flow_mask_ != 0)
```

Comment:
and this is my change, which was squash merged (ie a rebase of my branch and then a merge)